x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ql.optimizer;
+
+public class OptimizerUtils {
```

Comment:
The QL module is shared by multiple projects - unless there's a concrete need to have code in multiple places, work only in ESQL.
There's no other place where these methods are used so move them inside the optimizer rule. We can figure out in time if there's an opportunity for reuse.

Reply:
Yes, absolutely - already planned to move this into the esql project.